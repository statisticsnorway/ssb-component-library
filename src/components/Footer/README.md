Footer
========

> A wrapper for displaying a footer

### Usage
The component needs to be filled with content.

#### HTML

```html
<div class="ssb-footer-wrapper">
    <div class="top-row flex-row justify-space-between flex-wrap">
        {insert SSB logo with size h:44px w:248px}
        <button class="ssb-btn negative" onclick="window.scroll({ top: 0, behavior: 'smooth' })">
            <i>{feather.arrowUp}</i>
            Til toppen
        </button>
    </div>
    <div class="footer-content">
        <!-- Fill with content -->
    </div>
    <div class="bottom-row flex-row justify-space-between flex-wrap">
        <div class="global-links">
            <a class="ssb-link" href=" " target="" rel="">
                <span class="link-text">Statistisk sentralbyrå © 2019</span>
            </a>
            <a class="ssb-link" href=" " target="" rel="">
                <span class="link-text">A-Å</span>
            </a>
            <a class="ssb-link" href=" " target="" rel="">
                <span class="link-text">Nettstedskart</span>
            </a>
        </div>
        <div class="social-links">
            <a class="ssb-link with-icon" href=" " target="" rel="">
                <div class="icon-wrapper">
                    <i>{feather.facebook 24px}</i>
                </div>
            </a>
            <a class="ssb-link with-icon" href=" " target="" rel="">
                <div class="icon-wrapper">
                    <i>{feather.twitter 24px}</i>
                </div>
            </a>
            <a class="ssb-link with-icon" href=" " target="" rel="">
                <div class="icon-wrapper">
                    <i>{feather.linkedin 24px}</i>
                </div>
            </a>
            <a class="ssb-link with-icon" href=" " target="" rel="">
                <div class="icon-wrapper">
                    <i>{feather.rss 24px}</i>
                </div>
            </a>
        </div>
    </div>
</div>
```

#### React

```jsx harmony
<Footer>
  {/* Fill with content as below */}
    <div className="top-row flex-row justify-space-between flex-wrap">
        <SsbLogo alt="ssb-logo" />
        <Button negative onClick={() => window.scroll({ top: 0, behavior: 'smooth' })}>
            Til Toppen
        </Button>
    </div>
    <div className="footer-content">
        <!-- Fill with footer content -->
    </div>
    <div className="bottom-row flex-row justify-space-between flex-wrap">
        <div className="global-links">
            <Link href="https://www.ssb.no/" isExternal negative>Statistisk sentralbyrå © 2019</Link>
            <Link href="https://www.ssb.no/a-aa" isExternal negative>A-Å</Link>
            <Link href="https://www.ssb.no/nettstedskart" isExternal negative>Nettstedskart</Link>
        </div>
        <div className="social-links">
            <Link href="https://www.facebook.com/statistisksentralbyra/" isExternal negative icon={<Facebook size={24} />} />
            <Link href="https://twitter.com/ssbnytt" isExternal negative icon={<Twitter size={24} />} />
            <Link href="https://www.linkedin.com/company/statistics-norway/" isExternal negative icon={<Linkedin size={24} />} />
            <Link href="https://www.ssb.no/informasjon/rss" isExternal negative icon={<Rss size={24} />} />
        </div>
    </div>
</Footer>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children | node | All rendered content |
