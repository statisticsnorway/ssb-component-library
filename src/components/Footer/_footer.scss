.ssb-footer-wrapper {
  background: $ssb-dark-5;
  box-sizing: border-box;
  color: $ssb-white;
  padding: 36px 112px 58px;
  width: 100%;

  .top-row {
    border-bottom: 1px solid $ssb-white;
    margin-bottom: 36px;

    > * {
      margin-bottom: 36px;
    }
  }

  .bottom-row {
    margin-top: 100px;

    > * {
      margin-top: 20px;
    }

    .global-links {

      > * {
        margin-right: 20px;
      }
    }

    .social-links {

      > * {
        margin-left: 20px;
      }
    }
  }
}
