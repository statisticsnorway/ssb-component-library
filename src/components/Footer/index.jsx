import React from 'react';
import PropTypes from 'prop-types';

const Footer = ({ children }) => (
	<div className="ssb-footer-wrapper">
		{children}
	</div>
);

Footer.defaultProps = {};

Footer.propTypes = {
	children: PropTypes.node,
};

export default Footer;
