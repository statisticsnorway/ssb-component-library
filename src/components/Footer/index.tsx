import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface FooterProps {
  children?: ReactNode
  className?: string
}

const Footer: React.FC<FooterProps> = ({ children, className }) => (
  <div className={classNames('ssb-footer-wrapper', className)}>{children}</div>
)

export default Footer
