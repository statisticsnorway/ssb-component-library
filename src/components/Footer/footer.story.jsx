import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import {Â ChevronRight } from 'react-feather';
import Footer from './index';
import Link from '../Link';
import Title from '../Title';

storiesOf('Footer', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ width: '1200px' }}>
			<Footer>
				<div style={{ display: 'grid', gridTemplateColumns: 'repeat(5, auto)' }}>
					<div>
						<Title size={5} negative>Category</Title>
						<div style={{ display: 'flex', flexDirection: 'column' }}>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
						</div>
					</div>
					<div>
						<Title size={5} negative>Category</Title>
						<div style={{ display: 'flex', flexDirection: 'column' }}>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
						</div>
					</div>
					<div>
						<Title size={5} negative>Category</Title>
						<div style={{ display: 'flex', flexDirection: 'column' }}>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
						</div>
					</div>
					<div>
						<Title size={5} negative>Category</Title>
						<div style={{ display: 'flex', flexDirection: 'column' }}>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
						</div>
					</div>
					<div>
						<Title size={5} negative>Category</Title>
						<div style={{ display: 'flex', flexDirection: 'column' }}>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
							<Link icon={<ChevronRight />} href=" " negative>A link</Link>
						</div>
					</div>
				</div>
			</Footer>
		</div>
	));
