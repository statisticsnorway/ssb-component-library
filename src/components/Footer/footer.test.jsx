import React from 'react';
import { shallow } from 'enzyme';
import Footer from './index';

describe('Footer', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Footer><div>Content</div></Footer>);
		expect(wrapper).toMatchSnapshot();
	});
});
