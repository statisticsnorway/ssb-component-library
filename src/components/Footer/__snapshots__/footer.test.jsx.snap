// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Matches the snapshot 1`] = `
<div
  className="ssb-footer-wrapper"
>
  <div>
    Content
  </div>
</div>
`;
